beacon-chain/powchain/log_processing.go

```diff
@@ -155,7 +155,7 @@ func (s *Service) ProcessDepositLog(ctx context.Context, depositLog gethTypes.Lo
 	}
 
 	// We always store all historical deposits in the DB.
-	err = s.cfg.DepositCache.InsertDeposit(ctx, deposit, depositLog.BlockNumber, index, s.depositTrie.HashTreeRoot())
+	err = s.cfg.depositCache.InsertDeposit(ctx, deposit, depositLog.BlockNumber, index, s.depositTrie.HashTreeRoot())
```

Comment:
since this pattern moves us away from "config structs", should we start keeping these fields directly on the service object and get rid of the `.cfg` indirection?

Reply:
Nishant was concerned about polluting the service struct with all these fields. In this case we are using the `cfg` struct more as a bag to hold dependencies to the service. Maybe we can rename this struct to something else? Also, this is the first pass. In the second pass, I plan on cleaning up all these config structs, properties, and flags for each service